File: Release/WetiMatlabFunctions/PostProcessing/CalculateProcessResults.m
function ProcessResults = CalculateProcessResults(FrequencyResults,Statistics,PostProcessingConfig)

% return, if not defined or empty
if ~isfield(PostProcessingConfig,'CalculateProcessResults')||isempty(PostProcessingConfig.CalculateProcessResults)
    ProcessResults = struct();
    return
end

% get dimensions
Functions  	= PostProcessingConfig.CalculateProcessResults(:,1);
nFunctions	= size(Functions,1);

% initalization 
ProcessResults = struct();

% loop over functions
for iFunction = 1:nFunctions
    ThisFunction        = Functions{iFunction};
    ProcessResults      = ThisFunction(ProcessResults,FrequencyResults,Statistics);
end

end

File: Release/PythonFunctions/PostProcessing/CalculateProcessResults.py
def CalculateProcessResults(frequency_results, statistics, post_processing_config):
    if 'CalculateProcessResults' not in post_processing_config or not post_processing_config['CalculateProcessResults']:
        return {}

    functions = post_processing_config['CalculateProcessResults']
    process_results = {}

    for this_function in functions:
        process_results = this_function(process_results, frequency_results, statistics)

    return process_results